### ISSUE TITLE
Bad value for pe.number_of_exports on certain files

### ISSUE THREAD

**Issue by tlansec** [VirusTotal/yara#731]
Hi,

Recently whilst using yara with a DLL file with a high number of exports i found that the pe.number_of_exports field was undefined. 

0bad5ff96477fa009db357c42977ebf91c967d9b1df5d2b2c8f8f942a364960d

I was using the windows yara32.exe version 3.6.3, output below:

```C:\Users\malware\Desktop\analysis
λ cat _resources\always_false.rules
import "pe"
rule always_false
{
condition:
    false
}
C:\Users\malware\Desktop\analysis
λ yara32.exe --help
YARA 3.6.3, the pattern matching swiss army knife.
Usage: yara [OPTION]... RULES_FILE FILE | DIR | PID

Mandatory arguments to long options are mandatory for short options too.

  -t,  --tag=TAG                   print only rules tagged as TAG
  -i,  --identifier=IDENTIFIER     print only rules named IDENTIFIER
  -n,  --negate                    print only not satisfied rules (negate)
  -D,  --print-module-data         print module data
  -g,  --print-tags                print tags
  -m,  --print-meta                print metadata
  -s,  --print-strings             print matching strings
  -L,  --print-string-length       print length of matched strings
  -e,  --print-namespace           print rules' namespace
  -p,  --threads=NUMBER            use the specified NUMBER of threads to scan a directory
  -l,  --max-rules=NUMBER          abort scanning after matching a NUMBER of rules
  -d VAR=VALUE                     define external variable
  -x MODULE=FILE                   pass FILE's content as extra data to MODULE
  -a,  --timeout=SECONDS           abort scanning after the given number of SECONDS
  -k,  --stack-size=SLOTS          set maximum stack size (default=16384)
  -r,  --recursive                 recursively search directories
  -f,  --fast-scan                 fast matching mode
  -w,  --no-warnings               disable warnings
       --fail-on-warnings          fail on warnings
  -v,  --version                   show version information
  -h,  --help                      show this help and exit

Send bug reports and suggestions to: vmalvarez@virustotal.com.

C:\Users\malware\Desktop\analysis
λ yara32.exe _resources\always_false.rules 0bad5ff96477fa009db357c42977ebf91c967d9b1df5d2b2c8f8f942a364960d.MLWR -D | grep number_of_exports
        number_of_exports = UNDEFINED

C:\Users\malware\Desktop\analysis
λ```
The PE file in question should have 500 exports according to other tools such as PEStudio and VirusTotal:

https://www.virustotal.com/intelligence/search/?query=0bad5ff96477fa009db357c42977ebf91c967d9b1df5d2b2c8f8f942a364960d

Cheers,
Tom

**Comment by wxsBSD** [comment]
I know what's going on here. There is an invalid export data somewhere and we are returning early because of it instead of skipping it. I'll debug later today and have a patch up shortly.

**Comment by wxsBSD** [comment]
Fix is up. I decided it was better to try to continue processing the exports instead of returning early on the first one or break out of the loop.

**Comment by plusvic** [comment]
Fixed via #732
